src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -263,73 +366,137 @@ public String getRemote() {
+            for (GHPullRequest ghPullRequest : repo.getPullRequests(GHIssueState.OPEN)) {
+                int number = ghPullRequest.getNumber();
+                listener.getLogger().format("%n    Checking pull request %s%n", HyperlinkNote.encodeTo(ghPullRequest.getHtmlUrl().toString(), "#" + number));
+                boolean fork = !repo.getOwner().equals(ghPullRequest.getHead().getUser());
+                if (fork && !buildForkPRMerged && !buildForkPRUnmerged) {
+                    listener.getLogger().format("    Submitted from fork, skipping%n%n");
                     continue;
                 }
+                if (!fork && !buildOriginPRMerged && !buildOriginPRUnmerged) {
+                    listener.getLogger().format("    Submitted from origin repository, skipping%n%n");
+                    continue;
+                }
+                if (!fork) {
+                    originBranchesWithPR.add(ghPullRequest.getHead().getRef());
+                }
+                boolean trusted = isTrusted(repo, ghPullRequest);
+                if (!trusted) {
+                    listener.getLogger().format("    (not from a trusted source)%n");
+                }
+                for (boolean merge : new boolean[] {false, true}) {
+                    String branchName = "PR-" + number;
+                    if (merge && fork) {
+                        if (!buildForkPRMerged) {
+                            continue; // not doing this combination
+                        }
+                        if (buildForkPRUnmerged) {
+                            branchName += "-merged"; // make sure they are distinct
+                        }
+                    }
+                    if (merge && !fork) {
+                        if (!buildOriginPRMerged) {
+                            continue;
+                        }
+                        if (buildForkPRUnmerged) {
+                            branchName += "-merged";
+                        }
+                    }
+                    if (!merge && fork) {
+                        if (!buildForkPRUnmerged) {
+                            continue;
+                        }
+                        if (buildForkPRMerged) {
+                            branchName += "-unmerged";
+                        }
+                    }
+                    if (!merge && !fork) {
+                        if (!buildOriginPRUnmerged) {
+                            continue;
+                        }
+                        if (buildOriginPRMerged) {
+                            branchName += "-unmerged";
+                        }
+                    }
+                    PullRequestSCMHead head = new PullRequestSCMHead(ghPullRequest, branchName, merge, trusted);
+                    if (criteria != null) {
+                        SCMSourceCriteria.Probe probe = getProbe(branchName, "pull request", "refs/pull/" + number + (merge ? "/merge" : "/head"), repo, listener);
```

Comment:
Would be more precise to check whether the merge of the base branch head with the PR branch head contains a given file, etc., but this would be a lot more work, and is unlikely to differ from the above behavior.